Rubberduck.Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -54,18 +54,24 @@ public override string Description(IInspectionResult result)
 
         private string PromptForLocalVariableName(Declaration target, List<string> forbiddenNames)
         {
-            using (var view = _dialogFactory.Create(target.IdentifierName, target.DeclarationType.ToString(), forbiddenNames))
+            IAssignedByValParameterQuickFixDialog view = null;
+            try
             {
+                view = _dialogFactory.Create(target.IdentifierName, target.DeclarationType.ToString(), forbiddenNames);
                 view.NewName = GetDefaultLocalIdentifier(target, forbiddenNames);
                 view.ShowDialog();
-                
+
                 if (view.DialogResult == DialogResult.Cancel || !IsValidVariableName(view.NewName, forbiddenNames))
                 {
                     return string.Empty;
                 }
 
                 return view.NewName;
             }
+            finally
+            {
+                _dialogFactory.Release(view);
```

Comment:
Do all factories need an explicit `Release` call? Not that I'm religious about it, but per Seemann's "RRR" pattern there should only need to be one single call to `Release` the container in the app. Calling `Release` on the factory instance seems counter-intuitive to me. It's releasing the `view`?

Reply:
The thing why you call `factory.Release` with the auto-magic factories in CW is that CW tracks everything. If the `view` is registered in singleton scope, the release call does not do anything, at least I think so. However, if it is bound in transient scope, CW will dispose it, if disposable, and stop tracking it. Otherwise, it will hold on to a reference to the object until the container gets disposed.